Brian Holliday<br/>
Professor Masuda<br/>
Mathematical Modeling<br/>
21 October 2019<br/><br/><br/>

Project #2

Absract: 

For this problem we will be trying to find the optimal population for two species of whales. We will be trying to find the optimal population size that maximizes the population of each species while ensuring survival.<br/>

We will be using the five step model and we will model this as a unconstrained optimization problem.<br/>

After that we will be doing of sensitivity analysis of the rates to the growth rate. As well as a sensitivity analysis of the maximum sustainable populations to the growth rate

Ecologist use the following model to represent the growth process of two competing species, x and y:

$$\frac{dx}{dt} = (r1*x)*(1 - \frac{x}{K1})-a1*(x*y)$$

$$\frac{dy}{dt} = (r2*x)*(1 - \frac{y}{K2})-a2*(x*y)$$

$$r1 = 0.05$$
$$r2 = 0.08$$
$$K1 = 150,00$$
$$K2 = 400,00$$
$$a1 = 10^-8$$
$$a2 = 10^-8$$

The variables x and y represent the number in each popualation; the parameters ri represent the intrinsic growth rates of each species; Ki represents the maximum sustainable population in the absence of competition; and ai represents the effects of competition. Studies of the blue whale and fin whale populations have determined the following parameter values (t in years):


We are going to model this problem with the 5 step approach treating it as a unconstrated optimization problem.

Step 1: Identify the problem
    
   Determine the population levels x and y that maximize the number of new whales born year

Step 2: Make Assumptions <br/> <br/>
    Variables: <br/> <br/>
        x = population of blue whales <br/>
        y = population of fin whales <br/>
        r1 = growth rate of blue whales <br/>
        r2 = growth rate of fin whales <br/>
        K1 = highest possible population of blue whales <br/>
        K2 = highest possible population of fin whales <br/>
        a1 = Interaction coefficient of blue whales <br/>
        a2 = Interaction coefficient of fin whales <br/>
        T = Total popuation <br/> <br/>
    Assumptions:
        $$r1 = 0.05$$ <br/>
        $$r2 = 0.08$$ <br/>
        $$K1 = 150,000$$ <br/>
        $$K2 = 400,000$$ <br/>
        $$a1 = 10\exp{-8}$$ <br/>
        $$a2 = 10\exp{-8}$$ <br/>
        $$T = \frac{dx}{dt} + \frac{dy}{dt} $$ <br/>
        $$x > 0$$ <br/>
        $$y > 0$$

Step 3: Construct the model

In [1]:
r1 = 0.05
r2 = 0.08
K1 = 150000
K2 = 400000
a1 = 10**-8
a2 = a1
B(x,y) = (r1*x)*(1 - (x/K1)) - a1*(x*y)
F(x,y) = (r2*y)*(1 - (y/K2)) - a2*(x*y)
T(x,y) = B(x,y) + F(x,y)
#Lets solve the model

Step 4: Solve the Model

In [3]:
solve([diff(T(x,y),x)==0,diff(T(x,y),y)==0],x,y)
#Critical Points

[[x == (138000000/1997), y == (392500000/1997)]]

In [4]:
n(138000000/1997)
#Optimal Number of Blue Whales

69103.6554832248

In [5]:
n(392500000/1997)
#Optimal Number of Fin Whales

196544.817225839

In [6]:
T(69104, 196545)

9589.38407606667

We see for our model: <br/>

$$x = blue whales = 69104$$
$$y = fin whales = 196545$$
$$T(x,y) = 9589.38$$

The optimal population of blue whales is 69,104, while the optimal population of fin whales is 196,545. The optimal growth rate for the whales per year is 9589.

Step 5: Confirm the model <br/> <br/>
1) Sensitivity of growth rates to optimal population <br/>
2) Sensitivity of environmental carrying rates to optimal population <br/>



In [7]:
r1 = var('r1')
r2 = 0.08
K1 = 150000
K2 = 400000
a1 = 10**-8
a2 = a1
B(x,y,r1) = (r1*x)*(1 - (x/K1)) - a1*(x*y)
F(x,y) = (r2*y)*(1 - (y/K2)) - a2*(x*y)
T(x,y,r1) = B(x,y) + F(x,y)

In [8]:
solve([diff(T(x,y,r1),x)==0,diff(T(x,y,r1),y)==0],x,y)

[[x == 12000000*(250*r1 - 1)/(40000*r1 - 3), y == 7850000000*r1/(40000*r1 - 3)]]

In [9]:
dxdr1(x,r1) = diff((12000000*(250*r1 - 1)/(40000*r1 - 3)),r1)

In [51]:
dxdr1(x,r1) = diff((12000000*(250*r1 - 1)/(40000*r1 - 3)),r1)
Sx(x,r1) = dxdr1(x,r1) * (r1/x)
Sx(69104, .05)

0.0854538423283101

We get that the sensitivity of r1 to the blue whale popuation is 0.085<br/>
$$Sx(x,r1) = \frac{dy}{dr1}(r1) * \frac{r1}{x} = 0.085$$

In [14]:
dydr1(y,r1) = diff((7850000000*r1/(40000*r1 - 3)),r1)

In [15]:
Sy(y,r1) = dydr1(y,r1) * (r1/y)

In [16]:
Sy(196545,0.05)

-0.00150225198307145

We get that the sensitivity of r1 to the fin whale popuation is -0.0015<br/>
$$Sy(y,r1) = \frac{dy}{dr1}(r1) * \frac{r1}{y} = -0.0015$$

We will now compute the sensitivity of r2 to the growth rate

In [17]:
r1 = 0.05
r2 = var('r2')
K1 = 150000
K2 = 400000
a1 = 10**-8
a2 = a1
B(x,y) = (r1*x)*(1 - (x/K1)) - a1*(x*y)
F(x,y,r2) = (r2*y)*(1 - (y/K2)) - a2*(x*y)
T(x,y,r2) = B(x,y) + F(x,y,r2)

In [18]:
solve([diff(T(x,y,r2),x)==0,diff(T(x,y,r2),y)==0],x,y)

[[x == 1725000000*r2/(25000*r2 - 3), y == 2500000*(2000*r2 - 3)/(25000*r2 - 3)]]

In [19]:
dxdr2(x,r2) = diff((1725000000*r2/(25000*r2 - 3)),r2)
Sx(x,r2) = dxdr2(x,r2) * (r2/x)
Sx(69104, .08)

-0.00150224589061233

We get that the sensitivity of r2 to the blue whale popuation is -0.0015<br/>
$$Sy(x,r2) = \frac{dx}{dr2}(r2) * \frac{r2}{x} = -0.0015$$

In [20]:
dydr2(y,r2) = diff((2500000*(2000*r2 - 3)/(25000*r2 - 3)),r2)
Sy(y,r2) = dydr2(y,r2) * (r2/y)
Sy(196545,0.08)

0.0176060105022385

We get that the sensitivity of r2 to the fin whale popuation is 0.0176<br/>
$$Sy(y,r2) = \frac{dy}{dr2}(r2) * \frac{r2}{y} = 0.0176$$

In [21]:
r1 = 0.05
r2 = 0.08
K1 = var('K1')
K2 = 400000
a1 = 10**-8
a2 = a1
B(x,y,K1) = (r1*x)*(1 - (x/K1)) - a1*(x*y)
F(x,y) = (r2*y)*(1 - (y/K2)) - a2*(x*y)
T(x,y,K1) = B(x,y,K1) + F(x,y)

In [22]:
solve([diff(T(x,y,K1),x)==0,diff(T(x,y,K1),y)==0],x,y)

[[x == -46000000*K1/(K1 - 100000000), y == 2500000*(K1 - 8000000)/(K1 - 100000000)]]

In [23]:
dxdK1(K1) = diff((-46000000*K1/(K1 - 100000000)),K1)
Sx(x,K1) = dxdK1(K1) * (K1/x)
n(Sx(69104,150000))

1.00149726040822

We get that the sensitivity of K1 to the blue whale population is 1.0015<br/>
$$Sx(x,K2) = \frac{dx}{dK1}(K2) * \frac{K1}{x} = 1.0015$$

In [24]:
dydK1(K1) = diff((2500000*(K1 - 8000000)/(K1 - 100000000)),K1)
Sy(y,K1) = dydK1(K1) * (K1/y)
n(Sy(196545,150000))

-0.0176060105022386

We get that the sensitivity of K1 to the fin whale population is -0.0176<br/>
$$Sy(y,K1) = \frac{dx}{dK1}(K1) * \frac{K1}{x} = -0.0176$$

In [25]:
r1 = 0.05
r2 = 0.08
K1 = 150000
K2 = var('K2')
a1 = 10**-8
a2 = a1
B(x,y) = (r1*x)*(1 - (x/K1)) - a1*(x*y)
F(x,y,K2) = (r2*y)*(1 - (y/K2)) - a2*(x*y)
T(x,y,K2) = B(x,y) + F(x,y,K2)

In [26]:
solve([diff(T(x,y,K2),x)==0,diff(T(x,y,K2),y)==0],x,y)

[[x == 12000000*(K2 - 5000000)/(3*K2 - 800000000), y == -392500000*K2/(3*K2 - 800000000)]]

In [49]:
dxdK2(K2) = diff((12000000*(K2 - 5000000)/(3*K2 - 800000000)), K2)
Sx(x,K2) = dxdK2(K2) * (K2/x)
n(Sx(69104,400000))

-0.0854538423283102

We get that the sensitivity of K2 to the blue whale population is -0.085<br/>
$$Sx(x,K2) = \frac{dx}{dK2}(K2) * \frac{K2}{x} = -0.085$$

In [28]:
dydK2(K2) = diff((-392500000*K2/(3*K2 - 800000000)),K2)
Sy(y,K2) = dydK2(K2) * (K2/y)
n(Sy(196545,400000))

1.00150132204763

We get that the sensitivity of K2 to the fin whale population is 1.0015<br/>
$$Sy(y,K2) = \frac{dy}{dK2}(K2) * \frac{K2}{y} = 1.0015$$

Part D

Assuming that a1 = a2 = a, is it ever optimal for one species to become extinct?

Lets find the sensitivity of a to growth rate

In [29]:
r1 = 0.05
r2 = 0.08
K1 = 150000
K2 = 400000
a = var('a')
B(x,y,a) = (r1*x)*(1 - (x/K1)) - a*(x*y)
F(x,y,a) = (r2*y)*(1 - (y/K2)) - a*(x*y)
T(x,y,a) = B(x,y,a) + F(x,y,a)

In [30]:
solve([diff(T(x,y,a),x)==0,diff(T(x,y,a),y)==0],x,y)

[[x == 75000*(8000000*a - 1)/(15000000000000*a^2 - 1), y == 200000*(1875000*a - 1)/(15000000000000*a^2 - 1)]]

In [46]:
dxda(a) = (75000*(8000000*a - 1)/(15000000000000*a^2 - 1))
solve(dxda(a)==0,a)

[a == (1/8000000)]

In [42]:
n(1/8000000)

1.25000000000000e-7

We get that a has to be at least 1.25*10^-7 for the blue whale population to not be extinct

Let find a for y

In [44]:
dyda(a) = 200000*(1875000*a - 1)/(15000000000000*a^2 - 1)
solve(dyda(a)==0,a)

[a == (1/1875000)]

In [47]:
n(1/1875000)

5.33333333333333e-7

We get that a has to be at least 5.33*10^-7 for the fin whale population to not be extinct

Life is sustainable for both populations at $$a = 1.25e-7$$

Blue Whales become extinct at: $$a = 1.25e-7$$

Fin Whales become extinct at; $$a = 5.33e-7$$

Therefore in terms of a it is not optimal for one species to be extinct because the absolute value of the fin whales is more than blue whales